### PR TITLE
refactor: ♻️ add logical inclusion variables to classification logic

### LINKED ISSUE #401: Include all `has_` helper logic into final classification
Based on our discussion in the meeting, I think it would make it much clearer and easier to track what's going on it the `has_` helper logic variables are included in the final classification logic. So that we're clearly communicating (even just to each other) what actually needs to happen to get to that final state.

### PR DESCRIPTION
**Pull request by signekb**
## Description

This PR adds the logical inclusion variables to the `has_t1d` logic.

To be more explicit, I've also added separate `classify_t2d()` function with `has_t2d` logic that includes the logical inclusion variables as well. 

Let me know what you think about this implementation and if this is aligned with what was intended for #401.

As @Aastedet noted on Discord in regards to the current state of `main`, this PR also only survives the tests bc the edge case test is skipped.

Closes #401 

## Checklist

- [X] Ran `just run-all`

### CODE REVIEW

File: R/algorithm.R [hunk 1]
```diff
@@ -159,9 +159,15 @@ algorithm <- function() {
     has_t1d = list(
       register = NA,
       title = "Classifying type 1 diabetes status",
-      logic = "(has_only_insulin_purchases & has_any_t1d_primary_diagnosis) | (!has_only_insulin_purchases & has_majority_t1d_diagnoses & has_two_thirds_insulin & has_insulin_purchases_within_180_days)",
+      logic = "(has_diabetes_diagnosis OR has_podiatrist_service OR has_gld_purchase OR has_hba1c_over_threshold)>=2 AND (has_only_insulin_purchases & has_any_t1d_primary_diagnosis) OR (!has_only_insulin_purchases & has_majority_t1d_diagnoses & has_two_thirds_insulin & has_insulin_purchases_within_180_days)",
```
[the remark this answers is not in the document]
Comment (signekb) [reply R/algorithm.R]: Oh, you're right. My bad! 

```suggestion
      logic = "(has_diabetes_diagnosis OR has_podiatrist_service OR has_gld_purchase OR has_hba1c_over_threshold) AND (has_only_insulin_purchases & has_any_t1d_primary_diagnosis) OR (!has_only_insulin_purchases & has_majority_t1d_diagnoses & has_two_thirds_insulin & has_insulin_purchases_within_180_days)",
```

File: R/algorithm.R [hunk 2]
```diff
@@ -159,9 +159,15 @@ algorithm <- function() {
...
       comments = "The final classification for type 1 diabetes. Depends on all the previous steps to create these intermediate logical variables."
     ),
+    has_t2d = list(
+      register = NA,
+      title = "Classifying type 2 diabetes status",
+      logic = "(has_diabetes_diagnosis OR has_podiatrist_service OR has_gld_purchase OR has_hba1c_over_threshold)>=2 AND NOT has_t1d",
```
Comment (signekb) [inline R/algorithm.R]:
```suggestion
      logic = "(has_diabetes_diagnosis OR has_podiatrist_service OR has_gld_purchase OR has_hba1c_over_threshold) AND NOT has_t1d",
```

File: R/classify-diabetes.R [hunk 3]
```diff
@@ -214,3 +213,23 @@ classify_t1d <- function(data) {
       has_t1d = !!logic$has_t1d
     )
 }
+
+#' Classify those with type 2 diabetes.
```
[the remark this answers is not in the document]
Comment (lwjohnst86) [reply R/classify-diabetes.R]: No no, I think this is fine to have it clear. I just didn't recall what we said there. Likely was a misunderstanding, but I'm ok with this solution if @Aastedet is :relaxed:
